cmdutils/suite_test.go

```diff
@@ -0,0 +1,137 @@
+	eventuallyTimeout    = 3 * time.Second
+	consistentlyDuration = 1 * time.Second
+)
+
+var (
+	clients Clients
+)
+
+func TestMetalctl(t *testing.T) {
+	SetDefaultConsistentlyPollingInterval(pollingInterval)
+	SetDefaultEventuallyPollingInterval(pollingInterval)
+	SetDefaultEventuallyTimeout(eventuallyTimeout)
+	SetDefaultConsistentlyDuration(consistentlyDuration)
+	RegisterFailHandler(Fail)
+
+	RunSpecs(t, "Controller Suite")
+}
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
+
+	sourceEnv := &envtest.Environment{
+		CRDDirectoryPaths:     []string{filepath.Join("..", "config", "crd", "bases")},
+		ErrorIfCRDPathMissing: true,
+
+		// The BinaryAssetsDirectory is only required if you want to run the tests directly
+		// without call the makefile target test. If not informed it will look for the
+		// default path defined in controller-runtime which is /usr/local/kubebuilder/.
+		// Note that you must have the required binaries setup under the bin directory to perform
+		// the tests directly. When we run make test it will be setup and used automatically.
+		BinaryAssetsDirectory: filepath.Join("..", "bin", "k8s",
+			fmt.Sprintf("1.31.0-%s-%s", runtime.GOOS, runtime.GOARCH)),
+	}
+
+	sourceCfg, err := sourceEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(sourceCfg).NotTo(BeNil())
+
+	DeferCleanup(sourceEnv.Stop)
+
+	Expect(metalv1alpha1.AddToScheme(k8sSchema.Scheme)).NotTo(HaveOccurred())
+
+	//+kubebuilder:scaffold:scheme
+
+	clients.Source, err = client.New(sourceCfg, client.Options{Scheme: k8sSchema.Scheme})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(clients.Source).NotTo(BeNil())
+
+	targetEnv := &envtest.Environment{
+		CRDDirectoryPaths:     []string{filepath.Join("..", "config", "crd", "bases")},
+		ErrorIfCRDPathMissing: true,
+
+		// The BinaryAssetsDirectory is only required if you want to run the tests directly
+		// without call the makefile target test. If not informed it will look for the
+		// default path defined in controller-runtime which is /usr/local/kubebuilder/.
+		// Note that you must have the required binaries setup under the bin directory to perform
+		// the tests directly. When we run make test it will be setup and used automatically.
+		BinaryAssetsDirectory: filepath.Join("..", "bin", "k8s",
+			fmt.Sprintf("1.31.0-%s-%s", runtime.GOOS, runtime.GOARCH)),
+	}
+
+	// cfg is defined in this file globally.
+	targetCfg, err := targetEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(targetCfg).NotTo(BeNil())
+
+	DeferCleanup(targetEnv.Stop)
+
+	clients.Target, err = client.New(targetCfg, client.Options{Scheme: k8sSchema.Scheme})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(clients.Target).NotTo(BeNil())
+
+	By("Starting the registry server")
+	var mgrCtx context.Context
+	mgrCtx, cancel := context.WithCancel(context.Background())
+	DeferCleanup(cancel)
+	registryServer := registry.NewServer("localhost:30000")
```

Comment:
I would leave out the registry server out of this test suite.

Reply:
done ;)